### ISSUE TITLE
Spotlight Integration

### ISSUE THREAD

**Issue by ylian** [HappenApps/Quiver#91]
Currently there is no Spotlight integration for Quiver. You can't search for Quiver notes from Spotlight. This should be fixed.

You may not have Quiver open at all times. And not all your stuff is in Quiver. Spotlight search will provide a single place to search for all your stuff. Evernote seems to have Spotlight integration, but it also seems very buggy.

**Comment by ylian** [comment]
Unfortunately this doesn't work. I have implemented the Spotlight importer, but found out that Spotlight doesn't index files inside a file package. Since Quiver notes (.qvnote) are inside a Quiver notebook (.qvnotebook) which is a file package, Spotlight won't index the notes. If I move the .qvnote file outside of the notebook, Spotlight indexes it right away.

So either I have to change the file persistence scheme, or duplicate all the notes somewhere else. Neither seems to be a good idea.

Mark this as "WONT FIX" unless some other ideas come up in the future.

**Comment by ylian** [comment]
I have also tried using aliases and symlinks, but they didn't work either. Spotlight does index aliases, but doesn't use Quiver's Spotlight importer for aliases.

**Comment by ttscoff** [comment]
+1 on this. 

You have to export the spotlight-indexable content to caches and add an mdimporter that can associate the cached data with a specific note. You'd basically be caching the JSON in a temp folder for Spotlight's sake, but that's how Evernote and DEVONthink and others handle it.

**Comment by ylian** [comment]
@ttscoff Thanks. That seems to be the way to go. Reopen this.

**Comment by p0fi** [comment]
:+1: the spotlight search is a must have feature for me! :) 
Does something change in this with the deep links from OS X 10.11?

**Comment by ylian** [comment]
Probably coming in 3.1.

**Comment by p0fi** [comment]
great to hear! :)

**Comment by catesandrew** [comment]
The author of mailmate went through [a lot of trial and error](http://www.mail-archive.com/search?q=spotlight&l=mailmate%40lists.freron.com) to get spotlight working for emails. Probably worthwhile to go through them. He talked about `mdimporter` too, but now its a symlink from `HOME` folder to `~/Library`

**Comment by ylian** [comment]
@catesandrew Wow, thanks. That's very helpful.

**Comment by JMichaelTX** [comment]
> Spotlight search will provide a single place to search for all your stuff

@ylian:  Thanks for thinking of this, and considering it for implementation.
:+1:   I would find it very helpful.

**Comment by baryonyx5** [comment]
Another bump for spotlight integration.  After spending a week with Quiver, this is, for me, the only significant missing feature in this amazing app you've built.

**Comment by fsvehla** [comment]
:+1:

**Comment by habitualshaker** [comment]
me too. guessing that this would then allow quiver notes to be indexed by the likes of devonthink too?

**Comment by catesandrew** [comment]
devonthink doesn't use spotlight indexing. it has its own builtin system. anyways, you all could roll your own version pretty easily. you'll have to use source code formatting v3 <- map -> original line/col. quiver stores all notes in json blobs. go through each json blob, read its metadata, go through the contents, remove the 100-200 common words, stem the words with a stemmer, then store in some sort of make shift store you can use for a full search index. the source code formatting is responsible to map between the word searched for and its original line/col in the quiver note (since the notes are stored in one line). pretty sure quiver already has links to link to a document, use that link, append an anchor with the line/col and you're in business. just sayin _if_ its that important to you its feasible today. then you wouldn't be tied into spotlight since apple sends that info to their servers (for example) and could use your own, like launchbar, etc.

**Comment by habitualshaker** [comment]
i know it doesn't. was just hoping that whatever solution is put in place to enable spotlight indexing to happen would also allow devonthink to do the same.

**Comment by pacificera** [comment]
I'm evaluating quiver and this is a must-have feature for me to consider using it.  +1 request here.

**Comment by billmoritz** [comment]
Yes Please 👍

**Comment by KimMichaelJones** [comment]
A must have feature 👍

**Comment by versality** [comment]
Agree, spotlight search is a must

**Comment by SenorSamuel** [comment]
agree 👍

**Comment by JMichaelTX** [comment]
> Currently there is no Spotlight integration for Quiver. You can't search for Quiver notes from Spotlight. This should be fixed.

@ylian, you open this issue over a year and a half ago, so you clearly understand the need.
Has there been any progress on a design or solution?  Do you still plan to provide?

I really, really need this.  Spotlight finds all of my other files, notes, email, etc, including Evernote.

**Comment by ylian** [comment]
@JMichaelTX Please hang in there. I no longer have the luxury to work on Quiver full-time, so development is admittedly much slower. But I have every intention of continuing with this project and addressing all the issues.

**Comment by philosopherdog** [comment]
The development on Quiver is way too slow. Have you considered hiring some devs to get this product to its potential. The iOS client has completely languished. If you're too busy that's probably a sign to hire some devs. This product has a lot of potential but it will amount to not much if development isn't way swifter than the current pace.

**Comment by dkarter** [comment]
Another option: you can open source it to accept code contributions and still license it so that you get paid for it when people buy it on the App Store (or your own site, if you want to keep the 30% to yourself). I'm sure a lot of people wouldn't mind that (I would probably contribute).

**Comment by philosopherdog** [comment]
This app has so much potential. I feel like it's dead if someone doesn't step up and move this project along very soon. It's pretty close to amazing, but missing some key things especially syncing and a mobile client.

**Comment by ylian** [comment]
No need to worry.

**Comment by tflo** [comment]
Glad to hear that Spotlight is coming. This will increase the usefulness of Quiver by 100%.

**Comment by JMichaelTX** [comment]
### ylian commented on Apr 8, 2015  RE: Spotlight
> Currently there is no Spotlight integration for Quiver. You can't search for Quiver notes from Spotlight. This should be fixed.
> 
> You may not have Quiver open at all times. And not all your stuff is in Quiver. Spotlight search will provide a single place to search for all your stuff. Evernote seems to have Spotlight integration, but it also seems very buggy.

@ylian , as you stated two years ago, we definitely need **Spotlight** integration for Quiver.
Since then I have learned how to use, and depend many times a day on **Spotlight**.

Currently, I can do ONE **Spotlight** search for my scripts that returns:

  * All of my files that match
  * All of my Evernote Notes that match

In the **Spotlight** search I can filter for:

  * Characters in name of File or Note
  * Evernote Tags

I urgently need to add **Quiver** to Spotlight results.

BTW, I do NOT find the Spotlight search for Evernote to be "very buggy".  It works well for me.

**Comment by ylian** [comment]
@JMichaelTX No need to stress the importance of Spotlight integration. Allow me to push out more updates. Certain features require bigger changes, that's all.

**Comment by philosopherdog** [comment]
Glad to see development is resuming! This is a very brilliant app. I would love to see the ability to select multiple cells and move them around. Right now if you need to rearrange cells it's a huge pain.

**Comment by ylian** [comment]
@philosopherdog Yep I think there is a already a ticket filed for that.

**Comment by mapennell** [comment]
Just chiming in... long time Quiver user requesting spotlight integration. It's the one thing holding this app back in my opinion.

**Comment by pacificera** [comment]
My quest for perfect note taking has brought me back to see if spotlight support is available for quiver.  I'm asking because Quiver has almost everything else I want going for it.  But, unfortunately, after another year still no spotlight.  Guess I'll try back next year.

**Comment by KimMichaelJones** [comment]
@ylian, do you still plan to add this? Even a rough idea of when you expect to implement this would be a great help. I like the app but this keeps biting me in the ..

**Comment by gandalfsaxe** [comment]
I also really want to see this.

**Comment by yangyuw** [comment]
need this function very much

**Comment by pacificera** [comment]
Looks like another year will go by without spotlight integration.

**Comment by JMichaelTX** [comment]
> ### Looks like another year will go by without spotlight integration.

Yep.  And due mainly to the lack of Spotlight integration, **I have quit using Quiver.**
It is really too bad, because otherwise I like it a lot.  But I must be able to quickly find my snippets along with other related script files and info.  Spotlight gives me everything I need except for Quiver.

So I actually have moved on to using a combination of the macOS file system, with macOS Tags, and Evernote.  It is working very well.

Good luck to all.

**Comment by KimMichaelJones** [comment]
I too have given up on Quiver. I've found a combination of [SnippetLabs](https://www.renfei.org/snippets-lab/) and [MacDown](https://macdown.uranusjr.com) along with Spotlight search to work well for me.

**Comment by pacificera** [comment]
Thanks for the tip on snippetlabs.  I'll go with that.  Goodbye quiver!
